framework/src/modules/chain/synchronizer/utils.js

```diff
@@ -109,13 +121,15 @@ const restoreBlocksUponStartup = async (
 	storageModule,
 ) => {
 	// Get all blocks and find lowest height (next one to be applied)
-	const tempBlocks = await storageModule.entities.TempBlock.get();
-	const blockHighestHeight = tempBlocks.reduce((prev, current) =>
-		prev.height > current.height ? prev : current,
-	);
-	const blockLowestHeight = tempBlocks.reduce((prev, current) =>
-		prev.height < current.height ? prev : current,
+	const tempBlocks = await storageModule.entities.TempBlock.get(
```

Comment:
since we are getting the tempBlocks here, can't we just pass them to `restoreBlocks` function instead of `blocksModule` to avoid executing the same query again? Could you please take a look and refactor it if it's not too big?

Reply:
Let's do this in another issue once we move the logic to base synchronizer, this way we keep refactors isolated and we can review them further so this issue is fixed asap

Reply:
@limiaspasdaniel #4541